### PR TITLE
ensure element exists before access it

### PR DESCRIPTION
**Pull request by yshrsmz**
I got the following error when I created a layout without a header.

```
Uncaught TypeError: Cannot read properties of null (reading 'addEventListener')
    at main.min.js:1:2293
    at Array.forEach (<anonymous>)
    at HTMLDocument.<anonymous> (main.min.js:1:2271)
Uncaught TypeError: Cannot set properties of null (setting 'innerHTML')
    at setMode (main.min.js:1:1492)
    at window.onload (main.min.js:1:1803)
```

so added null checks to ensure that we use these elements when it actually exists.

<a href="https://gitpod.io/#https://github.com/sylhare/Type-on-Strap/pull/356"></a>

### CODE REVIEW

File: assets/js/partials/navbar.js [hunk 1]
```diff
@@ -8,16 +8,21 @@ document.addEventListener("DOMContentLoaded", function (event) {
...
+      }, false);
+    }
   });
 
   /*
    * Make the header images move on scroll
    */
   window.addEventListener('scroll', function () {
     const offset = -(window.scrollY || window.pageYOffset || document.body.scrollTop) / 3;
-    document.getElementById("main").style.backgroundPosition = '100% ' + (offset - 50) + 'px' + ', 0%, center top';
```
[the remark this answers is not in the document]
Comment (yshrsmz) [reply assets/js/partials/navbar.js:21]: no, it seems like uglifyjs does not support optional chaining.

somehow `pull?.addEventListener(...` does not throw error, but this line throws.

```
% gulp js
[19:08:31] Using gulpfile ~/repos/github.com/yshrsmz/Type-on-Strap/assets/gulpfile.js
[19:08:31] Starting 'js'...
GulpUglifyError: unable to minify JavaScript
Caused by: SyntaxError: Invalid assignment
File: /Users/a12897/repos/github.com/yshrsmz/Type-on-Strap/assets/js/partials/main.min.js
Line: 154
Col: 38
[19:08:31] The following tasks did not complete: js
[19:08:31] Did you forget to signal async completion?
```

[the remark this answers is not in the document]
Comment (yshrsmz) [reply assets/js/partials/navbar.js:21]: ah, I got it.
we can't do this

> It is invalid to try to assign to the result of an optional chaining expression:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#optional_chaining_not_valid_on_the_left-hand_side_of_an_assignment

File: assets/js/partials/navbar.js [hunk 2]
```diff
@@ -8,16 +8,21 @@ document.addEventListener("DOMContentLoaded", function (event) {
   const menu = document.querySelector('nav ul');
 
   ['click', 'touch'].forEach(function (e) {
-    pull.addEventListener(e, function () {
```
[the remark this answers is not in the document]
Comment (yshrsmz) [reply assets/js/partials/navbar.js:11]: fixed. [`8912a7a` (#356)](https://github.com/sylhare/Type-on-Strap/pull/356/commits/8912a7a0f79bfcabc6cc9693dc8be1152b308c55)

File: assets/js/partials/navbar.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (sylhare) [reply assets/js/partials/navbar.js:21]: ah alright true

**Review by sylhare — APPROVED** [review]

**Comment by yshrsmz** [comment]
thanks!

**Pull request merged**